### PR TITLE
Allows user to use the web service and specify a shape model with spiceinit

### LINKED ISSUE #1986: spiceinit web option not working with user defined shape
---

Author Name: **Stuart Sides** (@scsides)

---

The spiceinit application throws an error when a user specified shape model is used with the web=true option. I don't see why it couldn't just use the ellipse for testing the camera in the server then add the user's shape model back if it succeeded. This came from an external post:

"Forum link ":https://isis.astrogeology.usgs.gov/IsisSupport/index.php/topic,3919.0.html

To replicate this run spiceinit on a cube using a full path to one of the system dems:
@spiceinit from=ab.cub shape=user model=/usgs/cpkgs/isis3/data/base/dems/molaMarsPlanetaryRadius0005.cub web=t@

### PR DESCRIPTION
**Pull request by kaitlyndlee**
## Description

The spice server used to error when trying to use a user specified shape model. Refactored the code so that an ellipsoid is used for the web service, then overwrites the shape model keyword in the Kernels group with the user specified model.

## Related Issue

closes https://github.com/USGS-Astrogeology/ISIS3/issues/1986
## Motivation and Context

Allows user to specify a shape model while using the spice server.

## How Has This Been Tested?

Ran spiceinit on a test cube and DEM and made sure the output cube had the correct shape model in the Kernels group.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [x] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by kaitlyndlee** [comment]
Not sure if I should add a ctest for these changes since it uses the webs service.

**Review by jessemapel — COMMENTED** [review]
Change looks good, you should also check that you can still get a camera model after adding the shapemodel to the kernels group on the cube. Right now, if the shapemodel file is bad, the user won't find out until they run another app that hits the camera.

**Comment by kaitlyndlee** [comment]
@jessemapel Is line 679 enough with an error message update or did you have something else in mind?

**Comment by jessemapel** [comment]
Oh I didn't realize it was already in there 👍

**Comment by jessemapel** [comment]
For testing, we already have a spiceinit test that hits the web server. You should add one for this and just check that the shapemodel keyword in the kernels group is correct.

File: isis/src/base/apps/spiceinit/spiceinit.cpp [hunk 1]
```diff
@@ -662,12 +657,17 @@ namespace Isis {
         continue;
       }
     }
+    
+    if (QString(ui.GetString("SHAPE")).toLower() == "user") {
```
Comment (scsides) [inline isis/src/base/apps/spiceinit/spiceinit.cpp]: The toLower should not be necessary. When the UI/AML parameter is associated with a list of possible values (i.e. radio buttons) it always returns the uppercase version of the of the list element it uniquely matches. So change the litteral to uppercase and remove the toLower. Note, free form strings are not modified, but then why would we be doing a string comparison on a free form string?

File: isis/src/base/apps/spiceinit/spiceinit.cpp [hunk 2]
```diff
@@ -662,12 +657,17 @@ namespace Isis {
...
+      kernelsGroup["ShapeModel"] = ui.GetAsString("MODEL");
```
Comment (scsides) [inline isis/src/base/apps/spiceinit/spiceinit.cpp]: Not sure why GetAsString is being used here. Why not use getFile? The MODEL parameters is listed as type FILE. NOTE: Get as string is normally only used by the GUI to retrieve all values regardless of type.

**Review by scsides — CHANGES_REQUESTED** [review]

**Review by scsides — APPROVED** [review]

**Pull request merged**